export/export.go

```diff
@@ -0,0 +1,113 @@
+package export
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+	distrotypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	terra "github.com/terra-money/core/v2/app"
+)
+
+var IsAccountExportRunning = false
+
+func ExportAllAccounts(app *terra.TerraApp) error {
+	if IsAccountExportRunning {
+		return fmt.Errorf("exporting is still running")
+	}
+	IsAccountExportRunning = true
+	go exportWorker(app)
+	return nil
+}
+
+func ExportCirculatingSupply(app *terra.TerraApp) (sdktypes.Int, error) {
+	height := app.LastBlockHeight()
+	ctx := app.NewContext(true, tmproto.Header{Height: height})
+	time := time.Now()
+	totalVesting := sdktypes.NewInt(0)
+	app.AccountKeeper.IterateAccounts(ctx, func(account authtypes.AccountI) (stop bool) {
+		switch account.(type) {
+		case *vestingtypes.PeriodicVestingAccount:
+			v := account.(*vestingtypes.PeriodicVestingAccount)
+			totalVesting = totalVesting.Add(v.GetVestingCoins(time).AmountOf("uluna"))
+		default:
+			return false
+		}
+		return false
+	})
+	totalSupply, err := app.BankKeeper.TotalSupply(sdktypes.WrapSDKContext(ctx), &banktypes.QueryTotalSupplyRequest{})
+	if err != nil {
+		return sdktypes.Int{}, err
+	}
+	lunaTotalSupply := totalSupply.Supply.AmountOf("uluna")
+	communityPool, err := app.DistrKeeper.CommunityPool(sdktypes.WrapSDKContext(ctx), &distrotypes.QueryCommunityPoolRequest{})
+	if err != nil {
+		return sdktypes.Int{}, err
+	}
+	lunaCommunityPool := communityPool.Pool.AmountOf("uluna").TruncateInt()
+
+	return lunaTotalSupply.Sub(lunaCommunityPool).Sub(totalVesting), nil
+}
+
+func exportWorker(app *terra.TerraApp) {
```

Comment:
`runAccountExportWorker` would be more clearer

Reply:
+1